### PR TITLE
Add cpu_power

### PR DESCRIPTION
**Pull request by flgx16**
Calculates CPU draw in watts using voltage & current sensors.

Currently supports only `k10temp`.

### CODE REVIEW

**Comment by flgx16** [comment]
Cleaned it up, supports `k10temp` and `zenpower`.

**Comment by flgx16** [comment]
Added Intel RAPL support and removed `path.clear()` because I don't think it's needed there. Not sure but I think that RAPL energy reading is gonna be restricted soon because of a new CPU vulnerability (PLATYPUS).

File: src/cpu.h [hunk 1]
```diff
@@ -70,6 +139,7 @@ class CPUStats
    bool m_updatedCPUs = false; // TODO use caching or just update?
    bool m_inited = false;
    FILE *m_cpuTempFile = nullptr;
+   CPUPowerData* m_cpuPowerData = nullptr;
```
Comment (jackun) [inline src/cpu.h:142]: std::unique_ptr<CPUPowerData> m_cpuPowerData;

File: src/cpu.cpp [hunk 2]
```diff
@@ -240,6 +242,113 @@ bool CPUStats::UpdateCpuTemp() {
...
+
+bool CPUStats::UpdateCpuPower() {
+    if(!m_cpuPowerData)
+        return false;
+
+    int power = 0;
+
+    switch(m_cpuPowerData->source) {
+        case CPU_POWER_K10TEMP:
+            if (!get_cpu_power_k10temp(m_cpuPowerData, power)) return false;
```
Comment (jackun) [inline src/cpu.cpp:335]: `m_cpuPowerData.get()` and others

Comment (jackun) [inline src/cpu.cpp:335]: `m_cpuPowerData.get()`

**Comment by longusnickus** [comment]
Manjaro updated to Kernel 5.10. now i got max 30-40Watt at full load. It should be around 150Watt at full load on my 3700X

**Comment by flgx16** [comment]
Can you create a new issue for that please?

**Pull request merged**